examples/tv-casting-app/linux/simple-app.cpp

```diff
@@ -0,0 +1,153 @@
+    {
+        // generate a random Unique ID for this example app
+        for (size_t i = 0; i < sizeof(rotatingDeviceIdUniqueId); i++)
+        {
+            rotatingDeviceIdUniqueId[i] = chip::Crypto::GetRandU8();
+        }
+        rotatingDeviceIdUniqueIdSpan = chip::MutableByteSpan(rotatingDeviceIdUniqueId);
+    }
+
+    chip::MutableByteSpan * Get() { return &rotatingDeviceIdUniqueIdSpan; }
+};
+
+/**
+ * @brief Provides the ServerInitParams required to start the CastingApp, which in turn starts the Matter server
+ */
+class CommonCaseDeviceServerInitParamsProvider : public ServerInitParamsProvider
+{
+private:
+    // For this example, we'll use CommonCaseDeviceServerInitParams
+    chip::CommonCaseDeviceServerInitParams serverInitParams;
+
+public:
+    chip::ServerInitParams * Get()
+    {
+        CHIP_ERROR err = serverInitParams.InitializeStaticResourcesBeforeServerInit();
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr,
+                            ChipLogError(AppServer, "Initialization of ServerInitParams failed %" CHIP_ERROR_FORMAT, err.Format()));
+        return &serverInitParams;
+    }
+};
+
+int main(int argc, char * argv[])
+{
+    // Create AppParameters that need to be passed to CastingApp.Initialize()
+    AppParameters appParameters;
+    RotatingDeviceIdUniqueIdProvider rotatingDeviceIdUniqueIdProvider;
+    CommonCaseDeviceServerInitParamsProvider serverInitParamsProvider;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err            = InitCommissionableDataProvider(gCommissionableDataProvider, LinuxDeviceOptions::GetInstance());
+    VerifyOrReturnValue(
+        err == CHIP_NO_ERROR, 0,
+        ChipLogError(AppServer, "Initialization of CommissionableDataProvider failed %" CHIP_ERROR_FORMAT, err.Format()));
+    err = appParameters.Create(&rotatingDeviceIdUniqueIdProvider, &gCommissionableDataProvider,
+                               chip::Credentials::Examples::GetExampleDACProvider(),
+                               GetDefaultDACVerifier(chip::Credentials::GetTestAttestationTrustStore()), &serverInitParamsProvider);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, 0,
+                        ChipLogError(AppServer, "Creation of AppParameters failed %" CHIP_ERROR_FORMAT, err.Format()));
+
+    // Initialize the CastingApp
+    err = CastingApp::GetInstance()->Initialize(appParameters);
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, 0,
+                        ChipLogError(AppServer, "Initialization of CastingApp failed %" CHIP_ERROR_FORMAT, err.Format()));
+
+    // Initialize Linux KeyValueStoreMgr
+    chip::DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().Init(CHIP_CONFIG_KVS_PATH);
+
+    // Start the CastingApp
+    err = CastingApp::GetInstance()->Start();
+    VerifyOrReturnValue(err == CHIP_NO_ERROR, 0,
+                        ChipLogError(AppServer, "CastingApp::Start failed %" CHIP_ERROR_FORMAT, err.Format()));
+
+    chip::DeviceLayer::PlatformMgr().RunEventLoop();
```

Comment:
should this call go into the Start() function too?

Reply:
No. This should be called at the very end outside of the APIs, so the PlatformManager can process the submitted tasks and block for I/O.